SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -721,22 +761,42 @@ int64 USpatialActorChannel::ReplicateActor()
 		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
 	{
 		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
-		{		
+		{
+			const AActor* NetOwner = Actor->GetNetOwner();
+
+			uint64 HierarchyAuthorityTimestamp = AuthorityTimestamp;
+			if (NetOwner != nullptr)
+			{
+				GetLatestAuthorityChangeFromHierarchy(NetOwner, HierarchyAuthorityTimestamp);
+			}
+
+			const float TimeSinceReceivingAuthInSeconds = double(FPlatformTime::Cycles64() - HierarchyAuthorityTimestamp) * FPlatformTime::GetSecondsPerCycle64();
+			const float MigrationBackoffTimeInSeconds = 1.0f;
+
 			const VirtualWorkerId NewAuthVirtualWorkerId = NetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(*Actor);
-			if (NewAuthVirtualWorkerId != SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
+			if (TimeSinceReceivingAuthInSeconds < MigrationBackoffTimeInSeconds)
+			{
+				UE_LOG(LogSpatialActorChannel, Verbose, TEXT("Tried to change auth too early to %i for actor %s"), NewAuthVirtualWorkerId, *Actor->GetName());
+			}
+			else if (NewAuthVirtualWorkerId == SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
+			{
+				UE_LOG(LogSpatialActorChannel, Error, TEXT("Load Balancing Strategy returned invalid virtual worker for actor %s"), *Actor->GetName());
+			}
+			else
 			{
 				Sender->SendAuthorityIntentUpdate(*Actor, NewAuthVirtualWorkerId);
 
 				// If we're setting a different authority intent, preemptively changed to ROLE_SimulatedProxy 
 				Actor->Role = ROLE_SimulatedProxy;
 				Actor->RemoteRole = ROLE_Authority;
 
+				if (NetOwner != nullptr)
+				{
+					ForceReplicateOnActorHierarchy(NetDriver, NetOwner, Actor);
+				}
+
 				Actor->OnAuthorityLost();
```

Comment:
Have we considered the ordering of setting role to simulated proxy, triggered authority lost, and then doing the same on hierarchy? Might be fine but feel like there could be some issues here

Reply:
Yes, it would be more consistent.